jobbergate-api/jobbergateapi2/pagination.py

```diff
@@ -2,18 +2,92 @@
 Pagination feature for all endpoints.
 """
 
+from typing import Generic, List, Optional, TypeVar
+
+from pydantic import BaseModel
+from pydantic.generics import GenericModel
+from sqlalchemy import func, select
+
+from jobbergateapi2.storage import database
+
 
 class Pagination:
     """
     Basic pagination class.
     """
 
-    def __init__(self, limit: int = 10, skip: int = 0):
+    def __init__(self, page: Optional[int] = None, per_page: int = 10):
         """
-        Pagination constructor using default limit and skip offset.
+        Pagination constructor.
+
+        Raises an exception if page is negative values.
+        Raises an exception if per_page is less than 1.
+
+        :param page: The start page. If None, no pagination will be applied.
+        :param per_page: The number of items to include per page.
         """
-        self.limit = limit
-        self.skip = skip
+        if page is not None and page < 0:
+            raise ValueError("The page parameter must be greater than or equal to zero")
+        if per_page < 1:
+            raise ValueError("The per_page parameter must be greater than zero")
```

Comment:
what about using [conint](https://pydantic-docs.helpmanual.io/usage/types/#arguments-to-conint) from Pydantic instead of using if statements? I think the code gets cleaner and more lightweight

Reply:
Yeah, actually making this a Pydantic model makes some sense.

Reply:
Converted to a pydantic model in my latest commit.